src/wallet/wallet.cpp

```diff
@@ -4212,37 +4212,47 @@ bool CWallet::HasWalletDescriptor(const WalletDescriptor& desc) const
     return false;
 }
 
-std::shared_ptr<ScriptPubKeyMan> CWallet::AddWalletDescriptor(WalletDescriptor& desc, FlatSigningProvider& signing_provider, const std::string& label)
+std::shared_ptr<ScriptPubKeyMan> CWallet::AddWalletDescriptor(WalletDescriptor& desc, const FlatSigningProvider& signing_provider, const std::string& label, OutputType& output_type)
 {
     if (!IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS)) {
         return nullptr;
     }
 
     auto spk_manager = std::shared_ptr<DescriptorScriptPubKeyMan>(new DescriptorScriptPubKeyMan(std::bind(&CWallet::IsWalletFlagSet, this, std::placeholders::_1), std::bind(&CWallet::SetWalletFlag, this, std::placeholders::_1), std::bind(&CWallet::UnsetWalletFlagWithDB, this, std::placeholders::_1, std::placeholders::_2), std::bind(&CWallet::CanSupportFeature, this, std::placeholders::_1), std::bind(&CWallet::GetDisplayName, this), std::bind(&CWallet::SetMinVersion, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3), database, desc));
-    m_spk_managers[spk_manager->GetID()] = spk_manager;
-
-    // Save the descriptor & the privkeys from the descriptor to DB
-    spk_manager->WriteDescriptor();
-    for (const auto& entry : signing_provider.keys) {
-        const CKey& key = entry.second;
-        spk_manager->AddDescriptorKey(key, key.GetPubKey());
-    }
 
     // Top up key pool, the manager will generate new scriptPubKeys internally
     spk_manager->TopUp();
 
-    // Get the scriptPubKey (there should be only one as we disable labels for ranged descriptors)
-    // and apply the label if necessary
+    // Determine the output type by peeking into the first scriptPubKey (that we just generated with TopUp())
+    auto script_pub_keys = spk_manager->GetScriptPubKeys();
+    if (script_pub_keys.empty()) {
+        return nullptr;
+    }
+    const CScript& script = script_pub_keys.at(0);
+    if (Optional<OutputType> parsed_output_type = DetermineOutputType(script, signing_provider)) {
+        output_type = *parsed_output_type;
+    } else {
+        return nullptr;
```

Comment:
I don't think it makes sense to fail the import if we can't determine the `OutputType`, we just can't set it as active. People may want to import things that don't correspond to addresses but we still want to track those scripts.

Reply:
Ok, I'll update.